src/plugins/administrative_templates/ui/presentationbuilder.cpp

```diff
@@ -107,6 +107,95 @@ QHBoxLayout *createCaptions()
     return horizontalLayout;
 }
 
+QMap<std::string, QString> loadListFromRegistry(AbstractRegistrySource &source, const std::string &key, const std::string &prefix)
+{
+    QMap<std::string, QString> items;
+    std::vector<std::string> valueNames = source.getNonSpecialValueNames(key);
+
+    if(!prefix.empty())
+    {
+        // remove all valueNames(from return result), that doesn't have `prefix` prefix
+        valueNames.erase(std::remove_if(valueNames.begin(), valueNames.end(),
+                         [&prefix](const std::string& str)
+                         {
+                             return !(str.length() > prefix.length() && 
+                                     strncmp(str.c_str(), prefix.c_str(), prefix.length()));
+                         }), valueNames.end());
+    }
+
+    for (auto &valueName : valueNames) {
+        items[valueName] = 
+                source.getValue(key, valueName).value<QString>();
+    }
+    return items;
+}
+
+void cleanUpListInRegistry(AbstractRegistrySource &source, const std::string &key, const std::string &prefix = "")
+{
+    // small optimization
+    if (prefix.empty())
+    {
+        source.clearKey(key);
+    }
+
+    QString _prefix = QString::fromStdString(prefix);
+    std::vector<std::string> valueNames = source.getNonSpecialValueNames(key);
+    
+    // TODO: make case-insensitive.
+    // clean-up all values that contain `prefix` prefix (case-sensitive)
+    for (auto &value : valueNames) {
+        if (value.size() > prefix.size() && 
+            QString::fromUtf8(value.c_str(), prefix.size()).compare(_prefix, Qt::CaseInsensitive) == 0)
```

Comment:
Why you choose prefix to be case-insensitive in there? Is it something related to description of the prefix in documentation, or is it some other consideration?